op-proposer/proposer/l2_output_submitter.go

```diff
@@ -241,57 +241,45 @@ func (l *L2OutputSubmitter) Stop() {
 
 // FetchNextOutputInfo gets the block number of the next proposal.
 // It returns: the next block number, if the proposal should be made, error
-func (l *L2OutputSubmitter) FetchNextOutputInfo(ctx context.Context) (*eth.OutputResponse, bool, error) {
+func (l *L2OutputSubmitter) FetchNextOutputInfo(ctx context.Context) (*eth.OutputResponse, *big.Int, bool, error) {
 	cCtx, cancel := context.WithTimeout(ctx, l.networkTimeout)
 	defer cancel()
-	callOpts := &bind.CallOpts{
-		From:    l.txMgr.From(),
-		Context: cCtx,
-	}
-	nextCheckpointBlock, err := l.l2ooContract.NextBlockNumber(callOpts)
-	if err != nil {
-		l.log.Error("proposer unable to get next block number", "err", err)
-		return nil, false, err
-	}
+
 	// Fetch the current L2 heads
 	cCtx, cancel = context.WithTimeout(ctx, l.networkTimeout)
 	defer cancel()
 	status, err := l.rollupClient.SyncStatus(cCtx)
 	if err != nil {
 		l.log.Error("proposer unable to get sync status", "err", err)
-		return nil, false, err
+		return nil, nil, false, err
 	}
-	// Use either the finalized or safe head depending on the config. Finalized head is default & safer.
-	var currentBlockNumber *big.Int
+
+	// Use either the finalized or safe head depending on the config.
+	// Finalized head is default & safer.
+	currentBlockNumber := new(big.Int).SetUint64(status.FinalizedL2.Number)
```

Comment:
comment is now out of date & why is the behavior being changed?
We need to be able to propose safe but not finalized outputs for the devnet & tests (because clique does not finalize).